src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -10944,6 +10944,20 @@ private BoundConditionalAccess BindConditionalAccessExpression(ConditionalAccess
 
             var conditionalAccessBinder = new BinderWithConditionalReceiver(this, receiver);
             var access = conditionalAccessBinder.BindValue(node.WhenNotNull, diagnostics, BindValueKind.RValue);
+            Debug.Assert(access.Kind is not BoundKind.DeconstructionAssignmentOperator); // this shouldn't be possible syntactically
+            if (access.Kind
+                is BoundKind.AssignmentOperator
+                or BoundKind.CompoundAssignmentOperator
+                or BoundKind.NullCoalescingAssignmentOperator
+                or BoundKind.EventAssignmentOperator)
+            {
+                CheckFeatureAvailability(node.WhenNotNull, MessageID.IDS_FeatureNullConditionalAssignment, diagnostics);
```

Comment:
would it be possible to instead check this syntactically (while still in teh binder)?  It seems to be a single new syntactic structure, so it woudl be better than trying to explicitly enumerate all the semantic forms it morphs into.

Reply:
We could perhaps just negate the assertion from the else branch here. `if (access.Syntax is AssignmentExpressionSyntax) CheckFeatureAvailability(...)`.

I'll check what compiler team reviewers preference is before making the change. Thanks

Reply:
yes.  that would be my preference.  it's very very clear (since hte spec adds the new grammar production for c# next).  so we just check for that syntax production and check we're in the right version.  i strongly want this :)

Reply:
I still have this preference :-)

Reply:
@jjonescz @cston If there is no objection to Cyrus's suggestion I will try to make the change shortly.

Reply:
Done

Reply:
It occurs to me that it would also be reasonable to do the type check on `node.WhenNotNull` here, but, I don't feel it's important enough to change it.

Reply:
Resolvingt his out.  This looks great :)